test/dates/ranges.jl

```diff
@@ -482,10 +482,10 @@ function testmonthranges2(n)
 end
 testmonthranges2(100000)
 
-@test length(Dates.Year(1):Dates.Year(10)) == 10
+@test length(Dates.Year(1):Dates.Year(1):Dates.Year(10)) == 10
```

Comment:
I'm fine with requiring the user to specify the step, but this change should be documented.